### DISCUSSION TITLE
Is it possible to separate StackStorm components just like in the StackStorm flow diagram?

### DISCUSSION THREAD

**Discussion by aliasboink** [StackStorm/st2#5697] [Q&A]
The flow diagram I'm referring to is in the [overview](https://docs.stackstorm.com/overview.html). 
The question I have is whether or not the [components](https://docs.stackstorm.com/reference/ha.html) (like the ones there), i.e., st2actionrunner, st2rulesengine, etc. can be separated just like in the overview.
For instance, have entirely separated VM's with purely the `st2actionrunners` that can receive messages that were queued up by other VM's that serve as sensors. To my understanding you can more or less map the overview elements to different components (Worksers<->st2actionrunners, Rules Engine <-> st2rulesengine). This may be a bit of an oversimplification and I would appreciate a bit of clarification if possible, thanks in advance!
Ultimately the question is if one can separate the **components** akin to how they're depicted in the diagram to have entirely different VM's/entities doing actions and the others serve purely as sensors and so on.

**Comment by ip-sf** [comment]
Yes, they very much can be.

You can reference the [st2-docker](https://github.com/StackStorm/st2-docker) repo that deploy docker images on a per-binary basis (`st2rulesengine` runs in a separate container from `st2actionrunners`).

[Link to Stackstorm Dockerhub](https://hub.docker.com/u/stackstorm)

HTH
